test/scale/scale_tester.go

```diff
@@ -178,94 +186,117 @@ scaleInTest:
 }
 
 func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
-	statefulSets := []*resources.TestStatefulset{}
+	totalNumberOfPods := t.Replicas
+
+	nodes := nodeutil.ListAgentNodeNames(client, t.Pod.IsWindows)
+	skipper.SkipUnlessAtLeast(len(nodes), 2, "Need at least 2 agent nodes to verify the test case.")
+	midNode := len(nodes) / 2
+
+	testPod := resources.TestPod{
+		Client: client,
+	}
 
-	ginkgo.By("Initiating statefulset setup")
+	// uncordon the first half nodes
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
+	}
+
+	// cordon the second half nodes)
+	for _, node := range nodes[midNode:] {
+		node := node
+		framework.Logf("Cordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, true)
+	}
 
+	// start scale out test
+	startScaleOut := time.Now()
 	tStorageClass, storageCleanup := t.Pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
 	defer storageCleanup()
+	statefulSet, cleanupStatefulSet := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, &tStorageClass, map[string]string{"group": "azuredisk-scale-tester"})
+	defer cleanupStatefulSet(45 * time.Minute)
+
+	statefulSet.CreateWithoutWaiting()
+	ticker := time.NewTicker(time.Minute)
+	numberOfRunningPods1 := podutil.WaitForPodsWithMatchingLabelToRun(*scaleTestTimeout, ticker, totalNumberOfPods, client, namespace)
+	scaleOutCompleted := time.Since(startScaleOut).Minutes()
 
-	for i := 0; i < t.PodCount; i++ {
-		ss, cleanup := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, &tStorageClass, nil)
-		statefulSets = append(statefulSets, ss)
-		defer cleanup(45 * time.Minute)
+	// cordon the first half nodes where the running pods sit on
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Cordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, true)
 	}
 
-	ginkgo.By("Completed statefulset setup")
+	// uncordon the second nodes and start to create replica attachments
+	for _, node := range nodes[midNode:] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
+	}
 
-	ginkgo.By("Initiating statefulset deployment")
-	var wg sync.WaitGroup
-	start := time.Now()
-	for i := range statefulSets {
-		wg.Add(1)
-		go func(statefulset *resources.TestStatefulset) {
-			defer wg.Done()
-			defer ginkgo.GinkgoRecover()
+	roleReq, err := azureutils.CreateLabelRequirements(consts.RoleLabel, selection.Equals, string(azdiskv1beta2.ReplicaRole))
+	framework.ExpectNoError(err)
+	labelSelector := labels.NewSelector().Add(*roleReq)
 
-			statefulset.Create()
-		}(statefulSets[i])
-	}
+	ticker.Reset(time.Minute)
+	tickerCount := 0
+	timeout := time.After(time.Duration(*scaleTestTimeout) * time.Minute)
+	startCreateReplicaAtts := time.Now()
```

Comment:
We should be measuring the replica creation right after we uncordon the second group of nodes, around line 238